app/src/androidTest/java/com/github/emilehreich/bootcamp/MainActivityTest.kt

```diff
@@ -24,28 +23,22 @@ class MainActivityTest {
     val activityScenarioRule = ActivityScenarioRule(MainActivity::class.java)
 
     @Test
-    fun testWritingOfName() {
-        onView(ViewMatchers.withId(R.id.mainText)).perform(ViewActions.replaceText("Steve"))
-        closeSoftKeyboard()
-        onView(withText("Steve")).check(matches(isDisplayed()))
-    }
+    fun setAndGet() {
+        // 10.0.2.2 is the special IP address to connect to the 'localhost' of
+        // the host computer from an Android emulator.
+        database = Firebase.database
+        database.useEmulator("10.0.2.2", 9000)
 
-    @Test
-    fun testStartButton() {
-        Intents.init()
+        onView(withId(R.id.phoneBox)).perform(typeText("1234567890"))
+        onView(withId(R.id.mailBox)).perform(typeText("test@wouhou.ch"))
 
-        onView(withId(R.id.mainText)).perform(ViewActions.replaceText("Steve"))
-        closeSoftKeyboard()
+        onView(withId(R.id.setButton)).perform(click())
 
-        onView(withId(R.id.mainButton)).perform(click())
-        intended(
-            allOf(
-                hasComponent(GreetingActivity::class.java.name),
-                hasExtra("name", "Steve")
-            )
-        )
+        onView(withId(R.id.mailBox)).perform(typeText(""))
+        onView(withId(R.id.getButon)).perform(click())
 
-        Intents.release()
+        // check if displayed mail is correct
+        onView(withText("test@wouhou.ch")).check(matches(isDisplayed()))
     }
 
```

Comment:
The test works on my machine but nothing is displayed at the end 

And also is this one or two tests, because there are two different inputs for mailBox ?

Reply:
Looks good to me